### PR TITLE
Adding Github pipelines

### PR DESCRIPTION
**Pull request by hermeswaldemarin**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Java 17
```
Comment (marcio-absmartly) [inline .github/workflows/main.yml]: It's actually setting up java 8

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    tags:
+      - v*
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Java 17
```
Comment (marcio-absmartly) [inline .github/workflows/publish.yml]: Same here

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+          ORG_GRADLE_PROJECT_signingInMemoryKeyPassword: ${{ secrets.GPG_PASSWORD }}
+          ORG_GRADLE_PROJECT_mavenCentralUsername: ${{ secrets.MAVEN_CENTRAL_USERNAME }}
+          ORG_GRADLE_PROJECT_mavenCentralPassword: ${{ secrets.MAVEN_CENTRAL_PASSWORD }}
+          GRADLE_PROPERTIES: ${{ secrets.GRADLE_PROPERTIES }}
+          SECRET_FILE_DATA: ${{ secrets.SECRET_FILE_DATA }}
+        run: |
+          export GPG_TTY=$(tty)
+          mkdir -p ~/.gradle/
+          ln -s /usr/local/bin/gpg /usr/local/bin/gpg2
+          echo -n "$SECRET_FILE_DATA" | base64 --decode | gpg --import --passphrase $ORG_GRADLE_PROJECT_signingInMemoryKeyPassword --batch
```
Comment (marcio-absmartly) [inline .github/workflows/publish.yml]:
```suggestion
          echo -n "$SECRET_FILE_DATA" | base64 --decode | gpg --import --passphrase "$ORG_GRADLE_PROJECT_signingInMemoryKeyPassword" --batch
```

File: core-api/build.gradle [hunk 4]
```diff
@@ -8,6 +8,8 @@ description = """ABSmartly Java SDK"""
 
 def httpClientVersion = "5.1.3"
 
+def lombokVersion = "1.18.26"
```
Comment (marcio-absmartly) [inline core-api/build.gradle]: Is it safe to use Lombok with old Android versions?

Comment (hermeswaldemarin) [reply core-api/build.gradle]: Yeah, when I did it I forgot that was java 1.6 :(. Going to check.

**Comment by hermeswaldemarin** [comment]
Merged to branch feature/fix-analysis-issues with squash.

**Pull request closed without merging**